packages/ruby/lib/readme/request_queue.rb

```diff
@@ -0,0 +1,26 @@
+module Readme
+  class RequestQueue
+    attr_reader :queue
+
+    def initialize(buffer_length)
+      @queue = []
+      @buffer_length = buffer_length
+    end
+
+    def push(request)
+      @queue << request
+    end
+
+    def clear
+      @queue = []
+    end
+
+    def ready_to_send?
+      @queue.length >= @buffer_length
+    end
+
+    def to_json
+      "[#{@queue.join(", ")}]"
```

Comment:
I initially tried queueing `Payload` objects in the queue and then mapping over them with `to_json`, but I kept getting an error `IOError: not opened for reading`. I opted to parse the payload before passing it into the queue.

Reply:
Ah yes I think Rack closes all the IO streams when a request completes so you don't end up with dangling file handles